packages/chat/src/chat/chat.tsx

```diff
@@ -0,0 +1,323 @@
+  postMessage?: (
+    payload: TextPayload | ImagePayload,
+  ) => Promise<{ success: boolean; newMessages: MessageInterface[] }>
+  getMessages: (option: {
+    roomId: string
+    backward?: boolean
+    lastMessageId: number | string | null
+    req?: IncomingMessage
+  }) => Promise<MessageInterface[]>
+  getUnreadRoom?: (option: {
+    roomId: string
+    lastSeenMessageId: number
+  }) => Promise<HasUnreadOfRoomInterface>
+  room: RoomInterface
+}
+
+const defaultOnImageBubbleClick = (imageInfos: MetaDataInterface[]) => {
+  window.open(imageInfos[0].originalUrl, '_blank')
+}
+
+const Chat = ({
+  displayTarget,
+  userInfo,
+  postMessage,
+  getMessages,
+  getUnreadRoom,
+  room,
+
+  textBubbleFontSize,
+  textBubbleMaxWidthOffset,
+  mediaUrlBase,
+  cloudinaryName,
+  onRichBubbleButtonBeforeRouting,
+  onImageBubbleClick = defaultOnImageBubbleClick,
+  onTextBubbleClick,
+}: ChatProps) => {
+  const chatRoomRef = useRef<HTMLDivElement>(null)
+
+  const [
+    {
+      messages,
+      hasPrevMessage,
+      scrollY,
+      otherUnreadInfo,
+      lastMessageId,
+      firstMessageId,
+    },
+    dispatch,
+  ] = useReducer(ChatReducer, initialChatState)
+
+  const fetchJob = useMemo(() => new Polling(FETCH_INTERVAL_SECS * 1000), [])
+  const { os } = useUserAgentContext()
+  const isIos = useMemo(() => os.name === 'iOS', [os.name])
+
+  const updateUnread = useCallback(async () => {
+    if (!lastMessageId) {
+      return
+    }
+
+    const unreadRoomResult = await getUnreadRoom?.({
+      roomId: room.id,
+      lastSeenMessageId: lastMessageId,
+    })
+    const { hasUnread = false, others = [] } = unreadRoomResult || {}
+
+    dispatch({
+      action: ChatActions.UPDATE,
+      otherUnreadInfo: others,
+    })
+
+    return hasUnread
+  }, [room.id, lastMessageId])
+
+  const scrollDown = useCallback(() => {
+    if (chatRoomRef.current && chatRoomRef.current.parentElement) {
+      const height = chatRoomRef.current.getBoundingClientRect().height
+
+      chatRoomRef.current.parentElement.scrollTo(0, height)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (scrollY && chatRoomRef.current && chatRoomRef.current.parentElement) {
+      chatRoomRef.current.parentElement.scrollTo(
+        0,
+        chatRoomRef.current.getBoundingClientRect().height - scrollY,
+      )
+    }
+  }, [scrollY])
+
+  useEffect(() => {
+    fetchJob.run(() => pollingFetchJob())
+    return () => {
+      fetchJob.stop()
+    }
+  }, [lastMessageId]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    const chatListDiv = chatRoomRef.current
+    if (chatListDiv && isIos) {
+      chatListDiv.addEventListener('touchmove', () => closeKeyboard())
+      return () => {
+        chatListDiv.removeEventListener('touchmove', () => closeKeyboard())
+      }
+    }
+  }, [isIos])
+
+  useEffect(() => {
+    dispatch({
+      action: ChatActions.INIT,
+      messages,
+      lastMessageId: lastMessageId as number,
```

Comment:
```suggestion
      lastMessageId: Number(lastMessageId),
```
type assertion 대신 아래처럼 캐스팅이 나을 것 같아요

Reply:
[637e20f](https://github.com/titicacadev/triple-frontend/pull/2243/commits/637e20fffe128a1e4526b245591fbf077847c48d)
수정했습니다 🙇🏻